source/dotnet/Library/AdaptiveCards/AdaptiveBaseElement.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Xml.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace AdaptiveCards
+{
+    public class AdaptiveInternalID
+    {
+        public static uint Invalid = 0;
+        private static uint CurrentInternalID = 1;
+        private uint ID { get; }
+
+        public AdaptiveInternalID()
+        {
+            ID = Invalid;
+        }
+
+        private AdaptiveInternalID(uint id)
+        {
+            ID = id;
+        }
+
+        public static AdaptiveInternalID Current()
+        {
+            return new AdaptiveInternalID(CurrentInternalID);
+        }
+
+        public static AdaptiveInternalID Next()
+        {
+            CurrentInternalID++;
+
+            // handle overflow case
+            if (CurrentInternalID == AdaptiveInternalID.Invalid)
+            {
+                CurrentInternalID++;
+            }
+
+            return Current();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())
```

Comment:
>bj == null [](start = 17, length = 10)

nit: !obj

Reply:
apparently not allowed:

```
1>AdaptiveBaseElement.cs(47,17,47,21): error CS0023: Operator '!' cannot be applied to operand of type 'object'
```

---
In reply to: [278316768](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278316768) [](ancestors = 278316768)